src/Core/Components/Slider/FluentSlider.razor.cs

```diff
@@ -0,0 +1,124 @@
+// ------------------------------------------------------------------------
+// MIT License - Copyright (c) Microsoft Corporation. All rights reserved.
+// ------------------------------------------------------------------------
+
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+using Microsoft.AspNetCore.Components;
+using Microsoft.AspNetCore.Components.Web;
+using Microsoft.FluentUI.AspNetCore.Components.Extensions;
+
+namespace Microsoft.FluentUI.AspNetCore.Components;
+
+/// <summary>
+/// FluentSlider component, a slider control that allows users to select from a range of values.    
+/// </summary>
+/// <typeparam name="TValue"></typeparam>
+public partial class FluentSlider<TValue> : FluentInputBase<TValue>
+{
+    /// <summary>
+    /// Gets or sets the size for the slider.
+    /// </summary>
+    [Parameter]
+    public SliderSize? Size { get; set; }
+
+    /// <summary>
+    /// Gets or sets the slider's minimal value.
+    /// </summary>
+    [Parameter]
+    public TValue? Min { get; set; }
+
+    /// <summary>
+    /// Gets or sets the slider's maximum value.
+    /// </summary>
+    [Parameter, EditorRequired]
+    public TValue? Max { get; set; }
+
+    /// <summary>
+    /// Gets or sets the slider's step value.
+    /// </summary>
+    [Parameter]
+    public TValue? Step { get; set; }
+
+    /// <summary>
+    /// Gets or sets the orientation of the stacked components.
+    /// </summary>
+    [Parameter]
+    public Orientation Orientation { get; set; }
+
+    /// <summary>
+    /// Gets or sets the selection mode.
+    /// </summary>
+    [Parameter]
+    public SliderMode? Mode { get; set; }
+
+    /// <summary>
+    /// Gets or sets the content to be rendered inside the component.
+    /// </summary>
+    [Parameter]
+    public RenderFragment? ChildContent { get; set; }
+
+    /// <summary>
+    /// Gets or sets the value of the slider.
+    /// </summary>
+    protected override void OnParametersSet()
+    {
+        base.OnParametersSet();
+
+        if (typeof(TValue) != typeof(byte) &&
+           typeof(TValue) != typeof(int) &&
+           typeof(TValue) != typeof(long) &&
+           typeof(TValue) != typeof(float) &&
+           typeof(TValue) != typeof(double) &&
+           typeof(TValue) != typeof(decimal))
+        {
+            throw new InvalidOperationException("FluentSlider only supports numeric types.");
+        }
+    }
+
+    /// <summary>
+    /// Formats the value as a string. Derived classes can override this to determine the formatting used for <c>CurrentValueAsString</c>.
+    /// </summary>
+    /// <param name = "value">The value to format.</param>
+    /// <returns>A string representation of the value.</returns>
+    protected override string? FormatValueAsString(TValue? value)
```

Comment:
Due to the generic constraint (`where`), the override method accepts only a non-nullable TValue:
```csharp
protected override string? FormatValueAsString(TValue value)
{
    return Convert.ToString(value, System.Globalization.CultureInfo.InvariantCulture);
}
```